src/shared/models/asset.model.ts

```diff
@@ -0,0 +1,47 @@
+import type BigNumber from 'bignumber.js';
+
+import { WalletBlockchains } from './blockchain.model';
+import { Money } from './money.model';
+
+type WalletAssets = 'native' | 'ft' | 'nft';
+
+export interface FtMeta {
+  decimals: number;
+  name: string;
+  image_canonical_uri?: string;
+  image_uri?: string;
+  symbol: string;
+}
+
+// TODO: Revisit nft metadata
+// This appears to be used only in tests
+// Address property mixed casing
+// export interface NftMeta {
+//   count: string;
+//   key?: string;
+//   subCount?: string;
+//   total_sent: string;
+//   total_received: string;
+// }
+
+export interface BaseAsset {
+  balance: Money;
+  blockchain: WalletBlockchains;
+  hasMemo?: boolean;
+  name: string;
+  subtitle: string;
+  subBalance?: BigNumber;
+  type: WalletAssets;
+}
```

Comment:
In the exchange I worked on, we modelled with `Money` and `Currency` (effectively the same as an Asset we describe here). 

An interface that describes, say, Bitcoin, should contain info about the asset. Props here like: `blockchain`, `hasMemo`, `name`, `type` etc.
I don't think it should contain a balance. These are two separate things. What if we want to info about bitcoin, but not in a context of displaying an amount?

Either, we make a new interface for `Balance`, something like 
```ts
interface Balance {
  amount: Money;
  asset: Asset;
}
```

or, we just leave it to the balance list component to render the correct details from a `Money[]`

Reply:
Sure, makes sense, I can change.

Reply:
Thinking on this more, what is the difference b/w `Money` and `Currency`? I find that confusing.

Reply:
This doesn't make sense in our code though bc a `Balance` is not always a currency, `Money`, amount.

Reply:
I think this makes more sense...
```
export interface Currency extends BaseAsset {
  balance: Money;
  subBalance: Money;
  symbol: CryptoCurrencies;
}
````
And, remove balance and subBalance from the BaseAsset. Then a non-native fungible-token can be...
```
export interface StacksAssetFt extends BaseAsset {
  balance: BigNumber;
  canTransfer: boolean;
  contractAddress: string;
  contractName: string;
  subBalance: BigNumber;
  meta?: FtMeta;
}
```
Currency could also just be named `NativeAsset` above.

Reply:
This is actually confusing ...not sure your setup for `Money` and `CryptoCurrencies` works very well, does it? We can't manually add all transferable ft to the `CryptoCurrencies` type? I think you are referencing a past project that didn't have fungible tokens being created in addition to the blockchain gas asset?

Reply:
Thoughts on this...
```
type AssetTypes = 'native-ft' | 'non-native-ft' | 'nft';

export interface Asset {
  blockchain: Blockchains;
  hasMemo: boolean;
  name: string;
  type: AssetTypes;
}

export interface FtAsset extends Asset {
  canTransfer: boolean;
  contractAddress: string;
  contractName: string;
  meta: FtMeta;
}

export interface NftAsset extends Asset {
  contractAddress: string;
  contractName: string;
  meta: NftMeta;
}

export interface NativeAssetBalance {
  balance: Money;
  asset: Asset;
}

export interface FtAssetBalance {
  balance: Money;
  asset: FtAsset;
}

export interface NftAssetBalance {
  count: BigNumber;
  asset: NftAsset;
}
```
Flexible on the word `native`, but can't think of anything better.

Reply:
```ts
type AssetTypes = 'ft' | 'nft';

export interface Asset {
  blockchain: Blockchains;
  hasMemo: boolean;
  name: string;
  type: AssetTypes;
}

export interface FtAsset extends Asset {
  // 👇🏼 this creats a discriminating union
  type: 'ft';
  canTransfer: boolean;
  contractAddress: string; // stx specific, do we want to thus call it StxFtAsset?
  contractName: string;
  isNativeAsset: boolean;
  meta?: FtMeta;
}

export interface NftAsset extends Asset {
  type: 'nft';
  contractAddress: string;
  contractName: string;
  meta?: NftMeta; // if we can make it so that there is always meta data we'd have a much happier life
}

// this is how rosetta does it, fine to run with this, we'll just need an easy to
// initialise balances with all the asset metadata
export interface FtAssetBalance {
  balance: Money;
  asset: FtAsset; 
}

export interface NftAssetBalance {
  count: BigNumber;
  asset: NftAsset;
}
```